### DISCUSSION TITLE
Steps to integrate third-party feature detector

### DISCUSSION THREAD

**Discussion by ccaven** [OpenVSLAM-Community/openvslam_ros#178] [Ideas]
Hello! I appreciate your work in supporting open-source visual SLAM.
I am currently working on a (non-CUDA) GPU implementation of the ORB feature extractor, and would love to connect it to `stella_vslam_ros`. Here's what I think my TODO list is:
1. Add a new function in `stella_vslam` called `system::feed_mono_keypoints` / `system::feed_stereo_keypoints` which skips the ORB extraction and only runs `undistort_keypoints` then `convert_keypoints_to_bearings` and `assign_keypoints_to_grid`, etc.
2. Add a new subscription in `stella_vslam_ros` called `camera/keypoints` that then calls `system::feed_mono_keypoints`
3. In my own code, publish to `camera/keypoints`
You know the `stella_vslam` codebase better than I do, so if I'm missing anything / there is something I should know, please let me know. Thanks!

**Comment by ymd-stella** [comment]
Looks good to me. One thing to point out is that the type that represents keypoints is not in the ROS/ROS2 common message types. You must create a custom message.
